compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4011,7 +4011,7 @@ object Types {
       case tycon: TypeRef if tycon.symbol.isOpaqueAlias =>
         tycon.translucentSuperType.applyIfParameterized(args)
       case _ =>
-        superType
+        tryNormalize.orElse(superType)
```

Comment:
Why do this in translucentSuperType and not superType? At the very least it means the documentation of the base translucentSuperType in https://github.com/lampepfl/dotty/blob/f618dd66a93a962c99c22eb586dbc93d7d410988/compiler/src/dotty/tools/dotc/core/Types.scala#L1891-L1892 is now incorrect.

Reply:
I fixed the comment, which now also explains why we can't do it in `superType`.

Reply:
As far as I can see we did not normalize match types when computing the signatures and erasures of inherited members. That means we did not detect logical overrides, and that means we did not generate bridges where they would have been necessary. 

Match types that can be reduced directly in signatures were reduced. For instance in
```scala

  type M[X] = X match
    case Int => String
    case _ => X

  def test(x: M[Int]): Int = ???
```
the signature of `test` is `String => Int`.